Source/Core/Common/BitUtils.h

```diff
@@ -240,6 +241,53 @@ inline auto BitCastPtr(PtrType* ptr) noexcept -> BitCastPtrType<T, PtrType>
   return BitCastPtrType<T, PtrType>{ptr};
 }
 
+// Similar to BitCastPtr, but specifically for aliasing structs to arrays.
+template <typename ArrayType, typename T,
+          typename Container = std::array<ArrayType, sizeof(T) / sizeof(ArrayType)>>
+inline auto BitCastToArray(const T& obj) noexcept -> Container
+{
+  static_assert(sizeof(T) % sizeof(ArrayType) == 0,
+                "Size of array type must be a factor of size of source type.");
+  static_assert(std::is_trivially_copyable<T>(),
+                "BitCastToArray source type must be trivially copyable.");
+  static_assert(std::is_trivially_copyable<Container>(),
+                "BitCastToArray array type must be trivially copyable.");
+
+  Container result;
```

Comment:
This default constructs the `Container`, is this desirable? You may want to make this an aligned `std::byte` array.

Reply:
Uh, of course that's desirable. It's supposed to return an `std::array` of the type specified in the template.

Reply:
You memcpy data to it though, so essentially it default initialized for nothing.

Reply:
I don't believe you can avoid default initializing it.

Reply:
Good news is, default initialization is not zero initialization according to the docs, and leaves the newly allocated memory in some indeterminate state, which is ideal since we're writing over all of it anyways.